src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -584,7 +622,7 @@ private void StoreResultAndContinue(DkmEvaluationResult result, ArrayBuilder<Eva
             }
```

Comment:
I think this completionRoutine just adds something to the work list, so I don't believe it can throw.  If that's the case, then you probably don't need exception handling in the caller.